### PR TITLE
cassandra tweaking - to get conditional updates working in devenv

### PR DESCRIPTION
**Pull request by ankushgoyal**
Problem - Conditional Updates do not work with our Cassandra docker image.

Error seems to be - 
`
org.xerial.snappy.SnappyError: [FAILED_TO_LOAD_NATIVE_LIBRARY] null
    at org.xerial.snappy.SnappyLoader.load(SnappyLoader.java:239)
    at org.xerial.snappy.Snappy.<clinit>(Snappy.java:48)
    at org.xerial.snappy.SnappyOutputStream.<init>(SnappyOutputStream.java:79)
    at org.xerial.snappy.SnappyOutputStream.<init>(SnappyOutputStream.java:66)
    at org.apache.cassandra.net.OutboundTcpConnection.connect(OutboundTcpConnection.java:362)
    at org.apache.cassandra.net.OutboundTcpConnection.run(OutboundTcpConnection.java:150)
`

### CODE REVIEW

**Comment by ankushgoyal** [comment]
Test Case - 

`CREATE KEYSPACE IF NOT EXISTS test WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 };

CREATE TABLE IF NOT EXISTS test.connector_external_ids (
    external_id text,
    datasource_id text,
    last_action text,
    last_update timestamp,
    PRIMARY KEY ((external_id, datasource_id)))
    WITH compression = { 'sstable_compression' : 'LZ4Compressor' };

INSERT INTO connector_external_ids (external_id, datasource_id, last_action, last_update) VALUES ('a', 'b', 'add', dateOf(now()));

INSERT INTO connector_external_ids (external_id, datasource_id, last_action, last_update) VALUES ('a','b','add',dateOf(now())) IF NOT EXISTS;

UPDATE connector_external_ids
SET last_update = dateof(now())
WHERE external_id = 'a' and datasource_id = 'b'
IF last_update < dateof(now());`

File: cassandra.yaml [hunk 1]
```diff
@@ -509,7 +509,7 @@ compaction_preheat_key_cache: true
 # inter_dc_stream_throughput_outbound_megabits_per_sec:
 
 # How long the coordinator should wait for read operations to complete
-read_request_timeout_in_ms: 5000
+read_request_timeout_in_ms: 500000
```
Comment (jontg) [inline cassandra.yaml]: Why are we increasing this timeout to almost 10 minutes?

File: Dockerfile [hunk 2]
```diff
@@ -1,13 +1,16 @@
-FROM ingenieux/alpine-java8-supervisord:jdk-8u60-b27
+FROM java:openjdk-7-jdk-alpine
```
Comment (jontg) [inline Dockerfile]: Nice!
```
java                                                openjdk-7-jdk-alpine    de5c355bc76e        3 weeks ago          142.9 MB
ingenieux/alpine-java8-supervisord                  jdk-8u60-b27            11ac526a2565        9 months ago         215.9 MB
```

File: Dockerfile [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
+
+ARG MIRROR=http://archive.apache.org/dist/cassandra/2.0.17/apache-cassandra-2.0.17-bin.tar.gz
```
Comment (jontg) [inline Dockerfile]: I do not like this pattern.

File: Dockerfile [hunk 4]
```diff
@@ -1,13 +1,16 @@
...
 
 ## Make Alpine Linux resolve localhost
 RUN echo 'hosts: files mdns4_minimal [NOTFOUND=return] dns mdns4' >> /etc/nsswitch.conf
 
-RUN apk add --update wget && rm -rf /var/cache/apk/*
+RUN apk add --update --no-cache wget supervisor bash openssl curl && rm -rf /var/cache/apk/*
```
Comment (jontg) [inline Dockerfile]: with `--no-cache`, I believe you can remove the ` && rm -rf /var/cache/apk/*` at the end of this line.

File: Dockerfile [hunk 5]
```diff
@@ -1,13 +1,16 @@
...
 
 RUN mkdir -p /data /logs /init /opt/cassandra
-#RUN wget --no-check-certificate --progress=dot:mega -O - https://s3-us-west-1.amazonaws.com/relateiq-build-resources/apache-cassandra-2.0.11-bin.tar.gz | tar -zx -C /opt/cassandra --strip-components=1
-RUN wget --no-check-certificate --progress=dot:mega -O - http://archive.apache.org/dist/cassandra/2.0.17/apache-cassandra-2.0.17-bin.tar.gz | tar -zx -C /opt/cassandra --strip-components=1
+
+RUN wget -q -O - $MIRROR|tar -xzf - -C /opt/cassandra
```
Comment (jontg) [inline Dockerfile]: 👍 on `-q`!

**Comment by ankushgoyal** [comment]
Closing in favor of https://github.com/relateiq/docker-cassandra/pull/5

File: cassandra.yaml (hunk 1 above)
Comment (ankushgoyal) [inline cassandra.yaml]: hehe... pretty hacky... that snappy error was manifested as a rpc_timeout on client side... so was trying this out... but ofcourse this doesn't help

**Pull request closed without merging**